src/main/java/edu/ucla/library/prl/harvester/handlers/AddInstitutionHandler.java

```diff
@@ -1,38 +1,49 @@
 
 package edu.ucla.library.prl.harvester.handlers;
 
+import java.util.concurrent.CompletionStage;
+
 import org.apache.http.HttpStatus;
+import org.apache.solr.client.solrj.response.UpdateResponse;
 
 import edu.ucla.library.prl.harvester.Institution;
 import edu.ucla.library.prl.harvester.InvalidInstitutionJsonException;
 import edu.ucla.library.prl.harvester.MediaType;
 
+import io.ino.solrs.JavaAsyncSolrClient;
+
+import io.vertx.core.Future;
 import io.vertx.core.Vertx;
 import io.vertx.core.http.HttpHeaders;
 import io.vertx.core.http.HttpServerResponse;
 import io.vertx.core.json.JsonObject;
 import io.vertx.ext.web.RoutingContext;
+import io.vertx.sqlclient.Tuple;
 
 /**
  * A handler for adding institutions.
  */
-public final class AddInstitutionHandler extends AbstractRequestHandler {
+public final class AddInstitutionHandler extends AbstractSolrAwareWriteOperationHandler {
```

Comment:
Not a big deal, but I wonder whether `AbstractSolrAwareWriteOperationHandler` could just be named `AbstractWriteOperationHandler`. I get this change indicates it wasn't Solr aware before, but is this distinction necessary in the long run or would writes just be assumed to write to both locations?

Reply:
I thin the distinction is important because not all "write" operations (e.g., addJob) are what I'd consider Solr-aware (since Solr isn't modified during the request-response cycle of addJob, but rather: a job that has the potential to modify Solr is queued up).